### DISCUSSION TITLE
allow users to opt-in for MFA during register flow?

### DISCUSSION THREAD

**Discussion by paigexx** [keycloakify/keycloakify#398] [Q&A]
Is it possible to display the config totp view during user registration with the option to "Skip for now", allowing the user configure at a later time. We want to provide MFA as an optional layer of extra security for our users. What is the easiest way to achieve this functionality? 

The register flow I'm hoping to achieve is 
**1. User signs up (register.ftl)** 
**2. User verifies email (login-verify-email.ftl)**
**3. User shown the config totp view (login-config-totp.ftl) ->** 
_Once the user is shown the config totp view they have two options._ 
a.) Configure the totp 
b.) "Skip" the config totp view. If the user opts-out at this stage they can configure later with a redirect from our app settings page.
**4. Register complete and redirect to our application**

TIA

**Comment by garronej** [comment]
Hi @paigexx,  

Have you tried GPTing this?  
Sound like something ChatGPT4 could be able to help with.  

Beside, just a quick note, `register.ftl` should not be used in new project. `register-user-profile.ftl` is the more modern equivalent.

**Reply by paigexx** [reply]
Have not tried GPTing but good point on register.ftl. I will be migrating that to register-user-profile.ftl. Thanks!
